search/collector/topn.go

```diff
@@ -140,6 +142,20 @@ func (hc *TopNCollector) Collect(ctx context.Context, searcher search.Searcher,
 		DocumentMatchPool: search.NewDocumentMatchPool(backingSize+searcher.DocumentMatchPoolSize(), len(hc.sort)),
 	}
 
+	hc.dvReader, err = reader.DocValueReader(hc.neededFields)
+	if err != nil {
+		return err
+	}
+
+	if hc.updateFieldVisitor == nil {
```

Comment:
since this is now initialized earlier in Collect(), i think hc.updateFieldVisitor at this point in time is always going to be nil?

Reply:
sreekanth acting like a review response bot..hence this is not seen earlier..